### PR TITLE
Improve swap platforms rotating logo loading button to be responsive

### LINKED ISSUE #1191: Problem with scaling when loading on swap page
**Description**
Problem with scaling when loading on swap page (iphone 12 pro in screenshot) 

#### Definition of done
- [x] Description is created (Task creator)
- [x] Code is complete (Developer)
- [ ] Scope of test and what was done section is defined in task (Developer)
- [ ] Review passed (Code reviewer)
- [ ] Verified manually (QA)
- [ ] Test cases created/updated if necessary (QA)
- [ ] Change successfully merged
- [ ] Release notes added if needed

**Screenshots**

### PR DESCRIPTION
**Pull request by Diogomartf**
fixes #1191 

# Summary

fix split "finding best price" to be responsive in smaller screens.
 
**previously**
image

**Currently on mobile**
image

**Currently on desktop**
image

  # To Test

1. Make a swap on mobile
- [ ] check "finding best price" splits in one liner

### CODE REVIEW

**Review by wixzi — APPROVED** [review]
Tested and looks good. 👍🏼

**Comment by berteotti** [comment]
> Sometimes platform name is outside the button image image

Maybe is better to split into 2 rows?

**Comment by Diogomartf** [comment]
> Sometimes platform name is outside the button
> 
> image
> 
> image
> 
> 

I'll take a look at this and fix it. Thanks.

File: src/components/swap/SwapButtons/SwapButton.tsx [hunk 1]
```diff
@@ -83,22 +109,23 @@ export const SwapButton = ({
         ) : priceImpactSeverity > PRICE_IMPACT_HIGH && !isExpertMode ? (
           t('priceImpactTooHigh')
         ) : (
-          <>
-            {t('swapWith')}
+          <Flex alignItems="center">
+            <Text fontSize={'15px'}>{t('swapWith')}</Text>
```
Comment (berteotti) [inline src/components/swap/SwapButtons/SwapButton.tsx:113]:
```suggestion
            <Text fontSize="15px">{t('swapWith')}</Text>
```

Just to be cleaner :)

**Review by berteotti — APPROVED** [review]

**Comment by niemam29** [comment]
Tested and looks good to me!

File: src/components/swap/SwapButtons/SwapButton.tsx [hunk 2]
```diff
@@ -24,23 +24,49 @@ const StyledSwapLoadingButton = styled(ButtonPrimary)`
   background-image: linear-gradient(90deg, #4c4c76 19.74%, #292942 120.26%);
   cursor: 'wait';
   padding: 16px;
+
+  ${({ theme }) => theme.mediaWidth.upToExtraSmall`
+    display: grid;
+    gap: 8px;
+    justify-items: center;
```
Comment (mix1o) [inline src/components/swap/SwapButtons/SwapButton.tsx:31]: Grid is super cool but it's unnecessary here.

`StyledSwapLoadingButton`inherits styles from `Base` and it has `flex` property

```suggestion
    flex-direction: column
```

File: src/components/swap/SwapButtons/SwapButton.tsx [hunk 3]
```diff
@@ -24,23 +24,49 @@ const StyledSwapLoadingButton = styled(ButtonPrimary)`
...
-const StyledPlataformImage = styled.img`
-  margin-top: -3px;
-  margin-right: 6px;
+const RotatingLogo = styled.div`
+  position: relative;
+  display: flex;
+  justify-content: start;
+  align-items: center;
+  width: 112px;
+  height: 22px;
```
Comment (mix1o) [inline src/components/swap/SwapButtons/SwapButton.tsx:41]: Should we static set `width` and `height` ?
Consider using `min-height` and `min-width`

File: src/components/swap/SwapButtons/SwapButton.tsx [hunk 4]
```diff
@@ -24,23 +24,49 @@ const StyledSwapLoadingButton = styled(ButtonPrimary)`
...
 
-const StyledSwapButtonText = styled(Text)<{ width?: string }>`
+const StyledSwapButtonText = styled.div<{ width?: string }>`
   display: flex;
-  height: 16px;
-  white-space: pre-wrap;
-  width: ${({ width }) => (width ? `${width}px` : 'auto')};
+  justify-content: center;
+  align-items: center;
+  margin-left: 6px;
```
Comment (mix1o) [inline src/components/swap/SwapButtons/SwapButton.tsx:52]: 1. `width` is unused so you can remove it.
2. Correct me if I wrong but `flex` is useless here.

File: src/components/swap/SwapButtons/SwapButton.tsx (hunk 2 above)
Comment (Diogomartf) [reply src/components/swap/SwapButtons/SwapButton.tsx:31]: Makes sense Mikołaj, thanks!

File: src/components/swap/SwapButtons/SwapButton.tsx (hunk 3 above)
Comment (Diogomartf) [reply src/components/swap/SwapButtons/SwapButton.tsx:41]: I'll use `min-width` and `min-height`. Tks!

Comment (Diogomartf) [reply src/components/swap/SwapButtons/SwapButton.tsx]: Fixed!

File: src/components/swap/SwapButtons/SwapButton.tsx (hunk 2 above)
Comment (Diogomartf) [reply src/components/swap/SwapButtons/SwapButton.tsx]: Fixed!

**Review by MilanVojnovic95 — APPROVED** [review]
Tested and looks ok to me 👍

**Review by mix1o — APPROVED** [review]

**Pull request merged**